snmp/tests/test_e2e_core_metadata.py

```diff
@@ -0,0 +1,129 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from copy import deepcopy
+
+import pytest
+
+from . import common
+
+pytestmark = [pytest.mark.e2e, common.snmp_integration_only]
+
+
+def assert_network_devices_metadata(aggregator, events):
+    actual_events = aggregator.get_event_platform_events("network-devices-metadata", parse_json=True)
+    for event in actual_events:
+        # `collect_timestamp` depend on check run time and cannot be asserted reliably,
+        # we are replacing it with `0` if present
+        if 'collect_timestamp' in event:
+            event['collect_timestamp'] = 0
+    assert events == actual_events
+
+
+def test_e2e_core_metadata(dd_agent_check):
+    config = common.generate_container_instance_config([])
+    instance = config['instances'][0]
+    instance.update(
+        {
+            'community_string': 'f5-big-ip',
+            'loader': 'core',
+        }
+    )
+
+    aggregator = dd_agent_check(config, rate=False)
+
+    device_ip = instance['ip_address']
+    device_id = u'default:' + device_ip
+
+    events = [
```

Comment:
Would it be useful to test these usecases?
* event contains no devices, just interfaces
* event contains no interfaces, just devices

Reply:
Those cases are already testing in SNMP corecheck tests, I can be redundant to have e2e tests for that too.